{"cells":[{"cell_type":"code","source":["pip install tqdm"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"bb257c32-030e-40ad-8ceb-7ad10f7d95c0"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Python interpreter will be restarted.\nRequirement already satisfied: tqdm in /local_disk0/.ephemeral_nfs/envs/pythonEnv-619fdeac-a7fc-42cc-aeb1-fcca986154ee/lib/python3.8/site-packages (4.64.0)\nPython interpreter will be restarted.\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Python interpreter will be restarted.\nRequirement already satisfied: tqdm in /local_disk0/.ephemeral_nfs/envs/pythonEnv-619fdeac-a7fc-42cc-aeb1-fcca986154ee/lib/python3.8/site-packages (4.64.0)\nPython interpreter will be restarted.\n</div>"]}}],"execution_count":0},{"cell_type":"code","source":["pip install faker"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"85376602-ba93-4bf0-824f-1c8d34594634"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Python interpreter will be restarted.\nCollecting faker\n  Downloading Faker-13.11.0-py3-none-any.whl (1.5 MB)\nRequirement already satisfied: python-dateutil&gt;=2.4 in /databricks/python3/lib/python3.8/site-packages (from faker) (2.8.1)\nRequirement already satisfied: six&gt;=1.5 in /databricks/python3/lib/python3.8/site-packages (from python-dateutil&gt;=2.4-&gt;faker) (1.15.0)\nInstalling collected packages: faker\nSuccessfully installed faker-13.11.0\nPython interpreter will be restarted.\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Python interpreter will be restarted.\nCollecting faker\n  Downloading Faker-13.11.0-py3-none-any.whl (1.5 MB)\nRequirement already satisfied: python-dateutil&gt;=2.4 in /databricks/python3/lib/python3.8/site-packages (from faker) (2.8.1)\nRequirement already satisfied: six&gt;=1.5 in /databricks/python3/lib/python3.8/site-packages (from python-dateutil&gt;=2.4-&gt;faker) (1.15.0)\nInstalling collected packages: faker\nSuccessfully installed faker-13.11.0\nPython interpreter will be restarted.\n</div>"]}}],"execution_count":0},{"cell_type":"code","source":["pip install pymongo"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"ba97d173-2e9d-4a93-88d8-7277bdb17435"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Python interpreter will be restarted.\nCollecting pymongo\n  Downloading pymongo-4.1.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (480 kB)\nInstalling collected packages: pymongo\nSuccessfully installed pymongo-4.1.1\nPython interpreter will be restarted.\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Python interpreter will be restarted.\nCollecting pymongo\n  Downloading pymongo-4.1.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (480 kB)\nInstalling collected packages: pymongo\nSuccessfully installed pymongo-4.1.1\nPython interpreter will be restarted.\n</div>"]}}],"execution_count":0},{"cell_type":"code","source":["pip install dnspython"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"f8bb9899-b6d9-4e13-8ceb-a357ed4b01db"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Python interpreter will be restarted.\nCollecting dnspython\n  Downloading dnspython-2.2.1-py3-none-any.whl (269 kB)\nInstalling collected packages: dnspython\nSuccessfully installed dnspython-2.2.1\nPython interpreter will be restarted.\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Python interpreter will be restarted.\nCollecting dnspython\n  Downloading dnspython-2.2.1-py3-none-any.whl (269 kB)\nInstalling collected packages: dnspython\nSuccessfully installed dnspython-2.2.1\nPython interpreter will be restarted.\n</div>"]}}],"execution_count":0},{"cell_type":"code","source":["import numpy as np\nimport pandas as pd\nfrom tqdm import tqdm\nimport sys\nimport sklearn as sk\nfrom sklearn.ensemble import RandomForestClassifier\nfrom scipy import spatial\nimport collections\nfrom os.path import join, exists\n"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"c6cbdcc0-dccc-4e7c-912a-d7690b058566"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["# Read and process gene data"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"c3f1a361-4176-4b7d-a260-698fbd19f59d"}}},{"cell_type":"code","source":["def add_gene_location(df):\n    \"\"\"\n    Parsing gene dataframe attributes to find gene name and transforming gene location to good format.\n    Args:\n        df: Dataframe with gene information.\n    Return:\n        Gene df with added gene name and location.\n    \"\"\"\n    gene_data_df = df[df[\"type\"]==\"gene\"]\n    gene_data_df[\"gene_name\"] = gene_data_df.attributes.apply(lambda at: [x.split(\"=\")[-1] for x in str(at).split(\";\") if x.startswith(\"Name\")][0])\n    gene_data_df[\"location\"] = list(zip(gene_data_df.start, gene_data_df.end))\n    \n    return gene_data_df"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"86f76f44-a215-4a27-8d15-edc703e1b1a1"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["def process_gene_data_to_dict(gene_data_df):\n    \"\"\"\n    Grouping gene data (name and location) in a convenient format.\n    Args:\n        gene_data_df: Dataframe with gene information. Has to columns with location and gene name.\n    Return:\n        Dictionary with genes and their location.\n    \"\"\"\n    ordered_gene_dict = {}\n    for seqid, sub_df in gene_data_df.groupby('seqid'):\n        dict_seqid = dict(zip(sub_df.location, sub_df.gene_name))\n        dict_seqid_sorted = collections.OrderedDict(sorted(dict_seqid.items()))\n        ordered_gene_dict[seqid] = dict_seqid_sorted\n\n    return ordered_gene_dict"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"52c5cc2d-10fe-41d6-b7f1-4ac535687a74"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### Reading gene data csv as dataframe and parsing every row to fing gene name and gene location. Ordered_gene_dict is a dictionary with basic gene data: chromosome in which they are in as key and their name and exact location in genome as values."],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"b61f89e4-53ee-47f5-8b73-9cc1689c3601"}}},{"cell_type":"code","source":["column_names = ['seqid', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase', 'attributes']\nfile_path = \"/dbfs/FileStore/tables/Homo_sapiens_GRCh38_85_gff3.gz\"\ndf = pd.read_csv(file_path, compression='gzip',\n                 sep='\\t', comment='#', low_memory=False,\n                 header=None, names=column_names)\ngene_data_df = add_gene_location(df)\n\nordered_gene_dict = process_gene_data_to_dict(gene_data_df)"],"metadata":{"scrolled":true,"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"36612b85-c512-418d-83b4-ba8997995cf2"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">&lt;command-4400970693226260&gt;:10: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n  gene_data_df[&#34;gene_name&#34;] = gene_data_df.attributes.apply(lambda at: [x.split(&#34;=&#34;)[-1] for x in str(at).split(&#34;;&#34;) if x.startswith(&#34;Name&#34;)][0])\n&lt;command-4400970693226260&gt;:11: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n  gene_data_df[&#34;location&#34;] = list(zip(gene_data_df.start, gene_data_df.end))\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">&lt;command-4400970693226260&gt;:10: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n  gene_data_df[&#34;gene_name&#34;] = gene_data_df.attributes.apply(lambda at: [x.split(&#34;=&#34;)[-1] for x in str(at).split(&#34;;&#34;) if x.startswith(&#34;Name&#34;)][0])\n&lt;command-4400970693226260&gt;:11: SettingWithCopyWarning: \nA value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead\n\nSee the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n  gene_data_df[&#34;location&#34;] = list(zip(gene_data_df.start, gene_data_df.end))\n</div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["# Add gene name to patient snp and save to file"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"3671e8f3-7752-409a-8dc4-fd6c2220ec75"}}},{"cell_type":"code","source":["def find_gene_name_from_location(chromosome, position, ordered_gene_dict):\n    \"\"\"\n    Find gene names for chosen location in genome.\n    Args:\n        chromosome: chromosome of the gene.\n        position: exact position of a gene.\n        ordered_gene_dict: dictionary with gene data: their name and exact location in genome.\n    Return:\n        List of gene names.\n    \"\"\"\n    chosen_chromosome_dict = ordered_gene_dict[str(chromosome)]\n    genes = [v for k, v in chosen_chromosome_dict.items() if k[0] < position and k[1] > position]\n\n    return genes"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"c733d781-7efa-40ba-b035-0294d0237cb8"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["patient_data_path = \"/dbfs/FileStore/tables\"\npatient_list = [\"Child_1_Genome.csv\", \"Child_2_Genome.csv\", \"Child_3_Genome.csv\", \"Mother_Genome.csv\", \"Father_Genome.csv\"]"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"cf183d2b-9300-40bf-bc38-1e93b16b24d9"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### For every patient in patient_list add new column to csv with gene_name for every snp and save to csv. Skipped if modified csv already exist in patient_data_path."],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"58cfddd1-27c6-4088-9136-85f3058cfeed"}}},{"cell_type":"code","source":["for patient in patient_list:\n    patient_file = join(patient_data_path, patient)\n    new_patient_file_name = join(patient_data_path, patient.split(\".\")[0]+\"_geneNames.csv\")\n    \n    if not exists(new_patient_file_name):\n        patient_df = pd.read_csv(patient_file)\n        tqdm.pandas()\n        print(\"saving: \", new_patient_file_name)\n        patient_df['gene_name'] = patient_df.progress_apply(\n            lambda row: find_gene_name_from_location(row['chromosome'], float(row['position']), ordered_gene_dict), axis=1)\n\n        patient_df.to_csv(new_patient_file_name, index=False)"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"e2f1afe4-7c7a-4250-9cf5-d886805beb13"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["# Processing disease data"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"5b38bbab-314b-4997-b4a0-666f39e68491"}}},{"cell_type":"code","source":["disease_path = \"/dbfs/FileStore/tables/curated_gene_disease_associations.tsv\"\ndisease_df = pd.read_csv(disease_path, sep='\\t')"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"932c2d7e-efe1-446b-ae06-7a36f7b4ab89"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### Encode gene name to numeric representation and add as new column"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"cd32cbd3-36dd-4bb3-abfb-93e2af49d228"}}},{"cell_type":"code","source":["disease_df[\"geneSymbol_encoded\"] = disease_df[\"geneSymbol\"].astype('category').cat.codes"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"9f487468-3ba6-4af2-bc8e-b81ce4a6e4cc"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["disease_df"],"metadata":{"scrolled":true,"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"a4218d69-60f0-47e1-a194-a995eb3393d4"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Out[10]: </div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Out[10]: </div>"]}},{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>geneId</th>\n      <th>geneSymbol</th>\n      <th>DSI</th>\n      <th>DPI</th>\n      <th>diseaseId</th>\n      <th>diseaseName</th>\n      <th>diseaseType</th>\n      <th>diseaseClass</th>\n      <th>diseaseSemanticType</th>\n      <th>score</th>\n      <th>EI</th>\n      <th>YearInitial</th>\n      <th>YearFinal</th>\n      <th>NofPmids</th>\n      <th>NofSnps</th>\n      <th>source</th>\n      <th>geneSymbol_encoded</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>1</td>\n      <td>A1BG</td>\n      <td>0.700</td>\n      <td>0.538</td>\n      <td>C0019209</td>\n      <td>Hepatomegaly</td>\n      <td>phenotype</td>\n      <td>C23;C06</td>\n      <td>Finding</td>\n      <td>0.30</td>\n      <td>1.000</td>\n      <td>2017.0</td>\n      <td>2017.0</td>\n      <td>1</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>A1BG</td>\n      <td>0.700</td>\n      <td>0.538</td>\n      <td>C0036341</td>\n      <td>Schizophrenia</td>\n      <td>disease</td>\n      <td>F03</td>\n      <td>Mental or Behavioral Dysfunction</td>\n      <td>0.30</td>\n      <td>1.000</td>\n      <td>2015.0</td>\n      <td>2015.0</td>\n      <td>1</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>A2M</td>\n      <td>0.529</td>\n      <td>0.769</td>\n      <td>C0002395</td>\n      <td>Alzheimer's Disease</td>\n      <td>disease</td>\n      <td>C10;F03</td>\n      <td>Disease or Syndrome</td>\n      <td>0.50</td>\n      <td>0.769</td>\n      <td>1998.0</td>\n      <td>2018.0</td>\n      <td>3</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>2</td>\n      <td>A2M</td>\n      <td>0.529</td>\n      <td>0.769</td>\n      <td>C0007102</td>\n      <td>Malignant tumor of colon</td>\n      <td>disease</td>\n      <td>C06;C04</td>\n      <td>Neoplastic Process</td>\n      <td>0.31</td>\n      <td>1.000</td>\n      <td>2004.0</td>\n      <td>2019.0</td>\n      <td>1</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>2</td>\n      <td>A2M</td>\n      <td>0.529</td>\n      <td>0.769</td>\n      <td>C0009375</td>\n      <td>Colonic Neoplasms</td>\n      <td>group</td>\n      <td>C06;C04</td>\n      <td>Neoplastic Process</td>\n      <td>0.30</td>\n      <td>1.000</td>\n      <td>2004.0</td>\n      <td>2004.0</td>\n      <td>1</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>84033</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0002875</td>\n      <td>Cooley's anemia</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n    <tr>\n      <th>84034</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0005283</td>\n      <td>beta Thalassemia</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n    <tr>\n      <th>84035</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0019025</td>\n      <td>Hemoglobin F Disease</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n    <tr>\n      <th>84036</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0085578</td>\n      <td>Thalassemia Minor</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n    <tr>\n      <th>84037</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0271979</td>\n      <td>Thalassemia Intermedia</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n  </tbody>\n</table>\n<p>84038 rows × 17 columns</p>\n</div>","textData":null,"removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"htmlSandbox","arguments":{}}},"output_type":"display_data","data":{"text/html":["<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>geneId</th>\n      <th>geneSymbol</th>\n      <th>DSI</th>\n      <th>DPI</th>\n      <th>diseaseId</th>\n      <th>diseaseName</th>\n      <th>diseaseType</th>\n      <th>diseaseClass</th>\n      <th>diseaseSemanticType</th>\n      <th>score</th>\n      <th>EI</th>\n      <th>YearInitial</th>\n      <th>YearFinal</th>\n      <th>NofPmids</th>\n      <th>NofSnps</th>\n      <th>source</th>\n      <th>geneSymbol_encoded</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>1</td>\n      <td>A1BG</td>\n      <td>0.700</td>\n      <td>0.538</td>\n      <td>C0019209</td>\n      <td>Hepatomegaly</td>\n      <td>phenotype</td>\n      <td>C23;C06</td>\n      <td>Finding</td>\n      <td>0.30</td>\n      <td>1.000</td>\n      <td>2017.0</td>\n      <td>2017.0</td>\n      <td>1</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>A1BG</td>\n      <td>0.700</td>\n      <td>0.538</td>\n      <td>C0036341</td>\n      <td>Schizophrenia</td>\n      <td>disease</td>\n      <td>F03</td>\n      <td>Mental or Behavioral Dysfunction</td>\n      <td>0.30</td>\n      <td>1.000</td>\n      <td>2015.0</td>\n      <td>2015.0</td>\n      <td>1</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>A2M</td>\n      <td>0.529</td>\n      <td>0.769</td>\n      <td>C0002395</td>\n      <td>Alzheimer's Disease</td>\n      <td>disease</td>\n      <td>C10;F03</td>\n      <td>Disease or Syndrome</td>\n      <td>0.50</td>\n      <td>0.769</td>\n      <td>1998.0</td>\n      <td>2018.0</td>\n      <td>3</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>2</td>\n      <td>A2M</td>\n      <td>0.529</td>\n      <td>0.769</td>\n      <td>C0007102</td>\n      <td>Malignant tumor of colon</td>\n      <td>disease</td>\n      <td>C06;C04</td>\n      <td>Neoplastic Process</td>\n      <td>0.31</td>\n      <td>1.000</td>\n      <td>2004.0</td>\n      <td>2019.0</td>\n      <td>1</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>2</td>\n      <td>A2M</td>\n      <td>0.529</td>\n      <td>0.769</td>\n      <td>C0009375</td>\n      <td>Colonic Neoplasms</td>\n      <td>group</td>\n      <td>C06;C04</td>\n      <td>Neoplastic Process</td>\n      <td>0.30</td>\n      <td>1.000</td>\n      <td>2004.0</td>\n      <td>2004.0</td>\n      <td>1</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>84033</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0002875</td>\n      <td>Cooley's anemia</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n    <tr>\n      <th>84034</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0005283</td>\n      <td>beta Thalassemia</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n    <tr>\n      <th>84035</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0019025</td>\n      <td>Hemoglobin F Disease</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n    <tr>\n      <th>84036</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0085578</td>\n      <td>Thalassemia Minor</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n    <tr>\n      <th>84037</th>\n      <td>109580095</td>\n      <td>HBB-LCR</td>\n      <td>0.743</td>\n      <td>0.115</td>\n      <td>C0271979</td>\n      <td>Thalassemia Intermedia</td>\n      <td>disease</td>\n      <td>C16;C15</td>\n      <td>Disease or Syndrome</td>\n      <td>0.30</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>0</td>\n      <td>0</td>\n      <td>CTD_human</td>\n      <td>3605</td>\n    </tr>\n  </tbody>\n</table>\n<p>84038 rows × 17 columns</p>\n</div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["### Group genes for each disease"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"3be45ec2-f681-4273-b592-80a5246b621f"}}},{"cell_type":"code","source":["disease_grouped_df = disease_df.groupby('diseaseName', as_index=False).agg(list)\ndisease_grouped_df_filtered = disease_grouped_df[disease_grouped_df['geneSymbol'].apply(lambda x: len(x) > 400)]\n#get a list of unique disease names that are left after filtering\nunique_disease_filtered = disease_grouped_df_filtered['diseaseName'].unique()"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"fb9a52df-85d3-4c09-9196-272677259fc0"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["### One hot encode genes"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"5afa910b-e45d-4791-b00f-776365e17c7f"}}},{"cell_type":"code","source":["def create_word_embedding_disease(disease_df, add_weight=False):\n    \"\"\"\n    Parse disease_df to create word embeddings for disease and their genes.\n    Args:\n        disease_df: Dataframe with disease and gene information.\n        add_weight: Bool flag to add snp number as weight to word embeddings.\n    Return:\n        Dataframe with embedded genes and disease.\n    \"\"\"\n    one_hot_encoded_genes_df = pd.get_dummies(disease_df.geneSymbol)\n\n    #get disease name and number of snp\n    one_hot_encoded_genes_df[\"diseaseName\"] = disease_df[\"diseaseName\"]\n    one_hot_encoded_genes_df[\"NofSnp\"] = disease_df[\"NofSnps\"]\n\n    #get only diseases which have more genes than thresh\n    one_hot_encoded_genes_filtered_df = one_hot_encoded_genes_df[one_hot_encoded_genes_df[\"diseaseName\"].isin(unique_disease_filtered)]\n    \n    if add_weight:\n        #multipy encoded genes by nr of snps, WORKING ON UNGROUPED DF\n        gene_columns = one_hot_encoded_genes_filtered_df.columns.drop(['diseaseName', 'NofSnp'])\n        disease_names = one_hot_encoded_genes_filtered_df.diseaseName\n        one_hot_encoded_genes_filtered_df = one_hot_encoded_genes_filtered_df[gene_columns].multiply(one_hot_encoded_genes_filtered_df[\"NofSnp\"], axis=\"index\")\n        one_hot_encoded_genes_filtered_df['diseaseName'] = disease_names\n        \n    #group encoding by disease\n    one_hot_encoded_genes_grouped_df = one_hot_encoded_genes_filtered_df.groupby('diseaseName', as_index=False).agg(sum)\n    one_hot_encoded_genes_grouped_df.set_index(\"diseaseName\")\n    return one_hot_encoded_genes_grouped_df"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"db368c48-9f22-4285-bd4c-90d14de6f77c"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### Creating vectors with embedded genes for every disease. 1 if gene is responsible for a disease, 0 if it is not. Weighted version calculates also nr of snp: for every gene has nr_of_snp if gene is responsible for a disease, 0 if it is not."],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"f12dec6d-125b-40de-bb75-a342d79530a1"}}},{"cell_type":"code","source":["embedded_genes = create_word_embedding_disease(disease_df, False)\nembedded_genes_weight = create_word_embedding_disease(disease_df, True)"],"metadata":{"scrolled":true,"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"79522e5e-649a-4146-9c3c-ba2d1801aab1"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### A list of gene names that are left after filtering out gene dataframe"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"5cf284f1-09bc-4303-9e0f-5b08270f091c"}}},{"cell_type":"code","source":["available_genes = embedded_genes.columns\navailable_genes = available_genes.drop(['diseaseName'])"],"metadata":{"scrolled":true,"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"eaacc874-e9ca-497c-91df-1b6bd52d40cc"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### Preparing data for later use in model training"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"08928673-d606-4bc8-9b30-d28dfbd30230"}}},{"cell_type":"code","source":["y = embedded_genes.diseaseName.tolist()\nX = embedded_genes.loc[:, embedded_genes.columns != 'diseaseName'].to_numpy()"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"f30c9c69-ecea-4cbe-9e78-52dcad40d227"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["# Processing patient data"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"31f2af68-6935-4db6-ba25-a6051e433d0b"}}},{"cell_type":"code","source":["def do_embedding_patient_dict(patient_data, gene_data, add_weight=False):\n    \"\"\"\n    Parse patient data to create word embeddings for patients snp genes.\n    Args:\n        patient_data: Dictionary with snp genes and the number of times the occured.\n        gene_data: List with available genes.\n        add_weight: Bool flag to add snp number as weight to word embeddings.\n    Return:\n        Dictionary with embedded patient genes, only genes from available list.\n    \"\"\"\n    one_hot_encoded_patient = {}\n    for gene in gene_data:\n        if gene in patient_data.keys():\n            if add_weight:\n                one_hot_encoded_patient[gene] = patient_data[gene]\n            else:\n                one_hot_encoded_patient[gene] = 1\n        else:\n            one_hot_encoded_patient[gene] = 0\n    return one_hot_encoded_patient\n\ndef flatten(t):\n    return [item.strip(\"''\") for sublist in t for item in sublist if item]\n\ndef count_genes(gene_data):\n    \"\"\"\n    Returns a dict with gene names as keys and the number of snp that appeared in these genes as values.\n    \"\"\"\n    return dict(collections.Counter(gene_data))"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"771406b2-1380-4c3d-a4f5-d4174bec231c"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### List of files to process with patient data - csv files with added gene names to every snp"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"129520c2-335c-4c62-9e75-3827fe1dc8fc"}}},{"cell_type":"code","source":["patient_data_path = \"/dbfs/FileStore/tables\"\npatient_list = [\"Child_1_Genome_geneNames.csv\", \"Child_2_Genome_geneNames.csv\", \"Child_3_Genome_geneNames.csv\", \"Mother_Genome_geneNames.csv\", \"Father_Genome_geneNames.csv\"]"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"261c1983-d901-4d1d-83f5-de7ca73821a5"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### Processing genes for every patient. Returns a dictionary with every patient file name and a list of genes that have mutations. Based on available genes from gene data csv. If patient has a mutation in a gene that we don't know the name of it is not considered."],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"f1a82bcb-fe06-451d-9eb7-bc1688e30be3"}}},{"cell_type":"code","source":["patient_data_for_prediction = {}\nfor patient in patient_list:\n    named_patient_df = pd.read_csv(join(patient_data_path, patient))\n    \n    patient_gene_data = named_patient_df.gene_name.tolist()\n    patient_gene_data = [k.strip(\"[]\").split(\", \") for k in patient_gene_data]\n    patient_gene_data = flatten(patient_gene_data)\n    counted_patient_gene_data = count_genes(patient_gene_data)\n\n    encoded_patient_data = do_embedding_patient_dict(counted_patient_gene_data, available_genes)\n    encoded_patient_data_values = list(encoded_patient_data.values())\n    \n    patient_data_for_prediction[patient] = encoded_patient_data_values"],"metadata":{"scrolled":true,"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"6aebbfa8-8a1b-4f7f-8046-00b462d6b057"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">/databricks/python/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1) have mixed types.Specify dtype option on import or set low_memory=False.\n  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">/databricks/python/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (1) have mixed types.Specify dtype option on import or set low_memory=False.\n  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n</div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["# Random Forest"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"f2c58240-8738-4c2a-9d28-1c8e78b854fe"}}},{"cell_type":"markdown","source":["#### Creating and training Random Forest"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"b6bbefb9-4155-448a-b0ed-79684d825cda"}}},{"cell_type":"code","source":["clf=RandomForestClassifier(n_estimators=1000)\nclf.fit(X, y)"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"1e5cbd6e-ed98-4d83-8b3f-54eab5ee26ad"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Out[19]: RandomForestClassifier(n_estimators=1000)</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Out[19]: RandomForestClassifier(n_estimators=1000)</div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### Predicting probability of disease for every patient from patient_list"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"d8b7e450-0456-4289-866d-1acfa44bd557"}}},{"cell_type":"code","source":["predicted_disease = {}\nfor patient_path in patient_list:\n    encoded_patient_data_values = patient_data_for_prediction[patient_path]\n    y_pred = clf.predict([encoded_patient_data_values])\n    y_pred_prob=clf.predict_proba([encoded_patient_data_values])\n#     n = 5\n#     top_n = np.argsort(y_pred_prob)[:,:-n-1:-1]\n    top = np.argsort(y_pred_prob)[0][::-1]\n    \n    disease_prob = {}\n    for d in top:\n        disease_prob[y[d]] = round(y_pred_prob[0][d]*100, 2)\n    predicted_disease[patient_path] = disease_prob"],"metadata":{"scrolled":true,"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"63724693-e66c-433f-882d-f62137dc35e5"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["predicted_disease"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"c126d0f0-1bc7-4aa6-8304-91af72b43c52"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Out[21]: {&#39;Child_1_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 16.1,\n  &#39;Intellectual Disability&#39;: 11.4,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.6,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.4,\n  &#39;Prostatic Neoplasms&#39;: 8.4,\n  &#39;Liver carcinoma&#39;: 8.1,\n  &#39;Bipolar Disorder&#39;: 7.8,\n  &#39;Malignant neoplasm of prostate&#39;: 5.2,\n  &#39;Breast Carcinoma&#39;: 4.2,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.89,\n  &#39;Mammary Neoplasms&#39;: 1.6,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.4,\n  &#39;Hepatitis, Toxic&#39;: 1.32,\n  &#39;Drug-Induced Liver Disease&#39;: 1.31,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.28,\n  &#39;Mammary Carcinoma, Human&#39;: 0.71,\n  &#39;Mammary Neoplasms, Human&#39;: 0.59,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0},\n &#39;Child_2_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 14.9,\n  &#39;Intellectual Disability&#39;: 11.2,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.4,\n  &#39;Liver carcinoma&#39;: 9.1,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.1,\n  &#39;Prostatic Neoplasms&#39;: 8.2,\n  &#39;Bipolar Disorder&#39;: 7.9,\n  &#39;Malignant neoplasm of prostate&#39;: 5.5,\n  &#39;Breast Carcinoma&#39;: 4.7,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.91,\n  &#39;Hepatitis, Toxic&#39;: 1.57,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.53,\n  &#39;Drug-Induced Liver Disease&#39;: 1.36,\n  &#39;Mammary Neoplasms&#39;: 1.3,\n  &#39;Mammary Carcinoma, Human&#39;: 0.6,\n  &#39;Mammary Neoplasms, Human&#39;: 0.5,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0},\n &#39;Child_3_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 14.9,\n  &#39;Intellectual Disability&#39;: 11.2,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.4,\n  &#39;Liver carcinoma&#39;: 9.1,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.1,\n  &#39;Prostatic Neoplasms&#39;: 8.2,\n  &#39;Bipolar Disorder&#39;: 7.9,\n  &#39;Malignant neoplasm of prostate&#39;: 5.5,\n  &#39;Breast Carcinoma&#39;: 4.7,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.91,\n  &#39;Hepatitis, Toxic&#39;: 1.57,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.53,\n  &#39;Drug-Induced Liver Disease&#39;: 1.36,\n  &#39;Mammary Neoplasms&#39;: 1.3,\n  &#39;Mammary Carcinoma, Human&#39;: 0.6,\n  &#39;Mammary Neoplasms, Human&#39;: 0.5,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0},\n &#39;Mother_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 16.1,\n  &#39;Intellectual Disability&#39;: 11.4,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.6,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.4,\n  &#39;Prostatic Neoplasms&#39;: 8.4,\n  &#39;Liver carcinoma&#39;: 8.1,\n  &#39;Bipolar Disorder&#39;: 7.8,\n  &#39;Malignant neoplasm of prostate&#39;: 5.2,\n  &#39;Breast Carcinoma&#39;: 4.2,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.89,\n  &#39;Mammary Neoplasms&#39;: 1.6,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.4,\n  &#39;Hepatitis, Toxic&#39;: 1.32,\n  &#39;Drug-Induced Liver Disease&#39;: 1.31,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.28,\n  &#39;Mammary Carcinoma, Human&#39;: 0.71,\n  &#39;Mammary Neoplasms, Human&#39;: 0.59,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0},\n &#39;Father_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 16.1,\n  &#39;Intellectual Disability&#39;: 11.4,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.6,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.4,\n  &#39;Prostatic Neoplasms&#39;: 8.4,\n  &#39;Liver carcinoma&#39;: 8.1,\n  &#39;Bipolar Disorder&#39;: 7.8,\n  &#39;Malignant neoplasm of prostate&#39;: 5.2,\n  &#39;Breast Carcinoma&#39;: 4.2,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.89,\n  &#39;Mammary Neoplasms&#39;: 1.6,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.4,\n  &#39;Hepatitis, Toxic&#39;: 1.32,\n  &#39;Drug-Induced Liver Disease&#39;: 1.31,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.28,\n  &#39;Mammary Carcinoma, Human&#39;: 0.71,\n  &#39;Mammary Neoplasms, Human&#39;: 0.59,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0}}</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Out[21]: {&#39;Child_1_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 16.1,\n  &#39;Intellectual Disability&#39;: 11.4,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.6,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.4,\n  &#39;Prostatic Neoplasms&#39;: 8.4,\n  &#39;Liver carcinoma&#39;: 8.1,\n  &#39;Bipolar Disorder&#39;: 7.8,\n  &#39;Malignant neoplasm of prostate&#39;: 5.2,\n  &#39;Breast Carcinoma&#39;: 4.2,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.89,\n  &#39;Mammary Neoplasms&#39;: 1.6,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.4,\n  &#39;Hepatitis, Toxic&#39;: 1.32,\n  &#39;Drug-Induced Liver Disease&#39;: 1.31,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.28,\n  &#39;Mammary Carcinoma, Human&#39;: 0.71,\n  &#39;Mammary Neoplasms, Human&#39;: 0.59,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0},\n &#39;Child_2_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 14.9,\n  &#39;Intellectual Disability&#39;: 11.2,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.4,\n  &#39;Liver carcinoma&#39;: 9.1,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.1,\n  &#39;Prostatic Neoplasms&#39;: 8.2,\n  &#39;Bipolar Disorder&#39;: 7.9,\n  &#39;Malignant neoplasm of prostate&#39;: 5.5,\n  &#39;Breast Carcinoma&#39;: 4.7,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.91,\n  &#39;Hepatitis, Toxic&#39;: 1.57,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.53,\n  &#39;Drug-Induced Liver Disease&#39;: 1.36,\n  &#39;Mammary Neoplasms&#39;: 1.3,\n  &#39;Mammary Carcinoma, Human&#39;: 0.6,\n  &#39;Mammary Neoplasms, Human&#39;: 0.5,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0},\n &#39;Child_3_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 14.9,\n  &#39;Intellectual Disability&#39;: 11.2,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.4,\n  &#39;Liver carcinoma&#39;: 9.1,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.1,\n  &#39;Prostatic Neoplasms&#39;: 8.2,\n  &#39;Bipolar Disorder&#39;: 7.9,\n  &#39;Malignant neoplasm of prostate&#39;: 5.5,\n  &#39;Breast Carcinoma&#39;: 4.7,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.91,\n  &#39;Hepatitis, Toxic&#39;: 1.57,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.53,\n  &#39;Drug-Induced Liver Disease&#39;: 1.36,\n  &#39;Mammary Neoplasms&#39;: 1.3,\n  &#39;Mammary Carcinoma, Human&#39;: 0.6,\n  &#39;Mammary Neoplasms, Human&#39;: 0.5,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0},\n &#39;Mother_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 16.1,\n  &#39;Intellectual Disability&#39;: 11.4,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.6,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.4,\n  &#39;Prostatic Neoplasms&#39;: 8.4,\n  &#39;Liver carcinoma&#39;: 8.1,\n  &#39;Bipolar Disorder&#39;: 7.8,\n  &#39;Malignant neoplasm of prostate&#39;: 5.2,\n  &#39;Breast Carcinoma&#39;: 4.2,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.89,\n  &#39;Mammary Neoplasms&#39;: 1.6,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.4,\n  &#39;Hepatitis, Toxic&#39;: 1.32,\n  &#39;Drug-Induced Liver Disease&#39;: 1.31,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.28,\n  &#39;Mammary Carcinoma, Human&#39;: 0.71,\n  &#39;Mammary Neoplasms, Human&#39;: 0.59,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0},\n &#39;Father_Genome_geneNames.csv&#39;: {&#39;Malignant neoplasm of breast&#39;: 16.1,\n  &#39;Intellectual Disability&#39;: 11.4,\n  &#39;Colorectal Carcinoma&#39;: 9.7,\n  &#39;Schizophrenia&#39;: 9.6,\n  &#39;Liver Cirrhosis, Experimental&#39;: 9.4,\n  &#39;Prostatic Neoplasms&#39;: 8.4,\n  &#39;Liver carcinoma&#39;: 8.1,\n  &#39;Bipolar Disorder&#39;: 7.8,\n  &#39;Malignant neoplasm of prostate&#39;: 5.2,\n  &#39;Breast Carcinoma&#39;: 4.2,\n  &#39;Hepatitis, Drug-Induced&#39;: 1.89,\n  &#39;Mammary Neoplasms&#39;: 1.6,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 1.4,\n  &#39;Hepatitis, Toxic&#39;: 1.32,\n  &#39;Drug-Induced Liver Disease&#39;: 1.31,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 1.28,\n  &#39;Mammary Carcinoma, Human&#39;: 0.71,\n  &#39;Mammary Neoplasms, Human&#39;: 0.59,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 0.0}}</div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["# Vector similarity"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"fd698919-1f7c-489d-8a39-117e67e822e1"}}},{"cell_type":"markdown","source":["#### Calculating cosine vector similarity for every disease and every patient from patient_list"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"77e114b4-9c81-41fc-81a9-d5a99e8a58f1"}}},{"cell_type":"code","source":["predicted_disease = {}\nfor patient_path in patient_list:\n    encoded_patient_data_values = patient_data_for_prediction[patient_path]\n    disease_prob = {}\n    for i in range(len(y)):\n        similarity = 1 - spatial.distance.cosine(encoded_patient_data_values, X[i])\n        disease_prob[y[i]] = round(similarity*100, 2)\n    predicted_disease[patient_path] = disease_prob"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"ac36f3a5-ac12-428a-8ed1-ceb55ec74887"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["predicted_disease"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"867dc310-1e6d-4d36-aa0b-10fd72c01310"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Out[23]: {&#39;Child_1_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.02,\n  &#39;Breast Carcinoma&#39;: 3.16,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.75,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 13.82,\n  &#39;Colorectal Carcinoma&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 13.82,\n  &#39;Drug-Induced Liver Disease&#39;: 13.82,\n  &#39;Hepatitis, Drug-Induced&#39;: 13.82,\n  &#39;Hepatitis, Toxic&#39;: 13.82,\n  &#39;Intellectual Disability&#39;: 20.75,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.78,\n  &#39;Liver carcinoma&#39;: 3.38,\n  &#39;Malignant neoplasm of breast&#39;: 0.89,\n  &#39;Malignant neoplasm of prostate&#39;: 5.05,\n  &#39;Mammary Carcinoma, Human&#39;: 20.2,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.2,\n  &#39;Prostatic Neoplasms&#39;: 24.31,\n  &#39;Schizophrenia&#39;: 3.55},\n &#39;Child_2_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.04,\n  &#39;Breast Carcinoma&#39;: 3.17,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.98,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 14.05,\n  &#39;Colorectal Carcinoma&#39;: 1.55,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 14.05,\n  &#39;Drug-Induced Liver Disease&#39;: 14.05,\n  &#39;Hepatitis, Drug-Induced&#39;: 14.05,\n  &#39;Hepatitis, Toxic&#39;: 14.05,\n  &#39;Intellectual Disability&#39;: 21.12,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.74,\n  &#39;Liver carcinoma&#39;: 3.39,\n  &#39;Malignant neoplasm of breast&#39;: 0.88,\n  &#39;Malignant neoplasm of prostate&#39;: 5.06,\n  &#39;Mammary Carcinoma, Human&#39;: 20.24,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.24,\n  &#39;Prostatic Neoplasms&#39;: 24.36,\n  &#39;Schizophrenia&#39;: 3.56},\n &#39;Child_3_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.04,\n  &#39;Breast Carcinoma&#39;: 3.17,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.98,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 14.05,\n  &#39;Colorectal Carcinoma&#39;: 1.55,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 14.05,\n  &#39;Drug-Induced Liver Disease&#39;: 14.05,\n  &#39;Hepatitis, Drug-Induced&#39;: 14.05,\n  &#39;Hepatitis, Toxic&#39;: 14.05,\n  &#39;Intellectual Disability&#39;: 21.12,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.74,\n  &#39;Liver carcinoma&#39;: 3.39,\n  &#39;Malignant neoplasm of breast&#39;: 0.88,\n  &#39;Malignant neoplasm of prostate&#39;: 5.06,\n  &#39;Mammary Carcinoma, Human&#39;: 20.24,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.24,\n  &#39;Prostatic Neoplasms&#39;: 24.36,\n  &#39;Schizophrenia&#39;: 3.56},\n &#39;Mother_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.02,\n  &#39;Breast Carcinoma&#39;: 3.16,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.75,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 13.82,\n  &#39;Colorectal Carcinoma&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 13.82,\n  &#39;Drug-Induced Liver Disease&#39;: 13.82,\n  &#39;Hepatitis, Drug-Induced&#39;: 13.82,\n  &#39;Hepatitis, Toxic&#39;: 13.82,\n  &#39;Intellectual Disability&#39;: 20.75,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.78,\n  &#39;Liver carcinoma&#39;: 3.38,\n  &#39;Malignant neoplasm of breast&#39;: 0.89,\n  &#39;Malignant neoplasm of prostate&#39;: 5.05,\n  &#39;Mammary Carcinoma, Human&#39;: 20.2,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.2,\n  &#39;Prostatic Neoplasms&#39;: 24.31,\n  &#39;Schizophrenia&#39;: 3.55},\n &#39;Father_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.02,\n  &#39;Breast Carcinoma&#39;: 3.16,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.75,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 13.82,\n  &#39;Colorectal Carcinoma&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 13.82,\n  &#39;Drug-Induced Liver Disease&#39;: 13.82,\n  &#39;Hepatitis, Drug-Induced&#39;: 13.82,\n  &#39;Hepatitis, Toxic&#39;: 13.82,\n  &#39;Intellectual Disability&#39;: 20.75,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.78,\n  &#39;Liver carcinoma&#39;: 3.38,\n  &#39;Malignant neoplasm of breast&#39;: 0.89,\n  &#39;Malignant neoplasm of prostate&#39;: 5.05,\n  &#39;Mammary Carcinoma, Human&#39;: 20.2,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.2,\n  &#39;Prostatic Neoplasms&#39;: 24.31,\n  &#39;Schizophrenia&#39;: 3.55}}</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Out[23]: {&#39;Child_1_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.02,\n  &#39;Breast Carcinoma&#39;: 3.16,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.75,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 13.82,\n  &#39;Colorectal Carcinoma&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 13.82,\n  &#39;Drug-Induced Liver Disease&#39;: 13.82,\n  &#39;Hepatitis, Drug-Induced&#39;: 13.82,\n  &#39;Hepatitis, Toxic&#39;: 13.82,\n  &#39;Intellectual Disability&#39;: 20.75,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.78,\n  &#39;Liver carcinoma&#39;: 3.38,\n  &#39;Malignant neoplasm of breast&#39;: 0.89,\n  &#39;Malignant neoplasm of prostate&#39;: 5.05,\n  &#39;Mammary Carcinoma, Human&#39;: 20.2,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.2,\n  &#39;Prostatic Neoplasms&#39;: 24.31,\n  &#39;Schizophrenia&#39;: 3.55},\n &#39;Child_2_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.04,\n  &#39;Breast Carcinoma&#39;: 3.17,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.98,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 14.05,\n  &#39;Colorectal Carcinoma&#39;: 1.55,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 14.05,\n  &#39;Drug-Induced Liver Disease&#39;: 14.05,\n  &#39;Hepatitis, Drug-Induced&#39;: 14.05,\n  &#39;Hepatitis, Toxic&#39;: 14.05,\n  &#39;Intellectual Disability&#39;: 21.12,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.74,\n  &#39;Liver carcinoma&#39;: 3.39,\n  &#39;Malignant neoplasm of breast&#39;: 0.88,\n  &#39;Malignant neoplasm of prostate&#39;: 5.06,\n  &#39;Mammary Carcinoma, Human&#39;: 20.24,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.24,\n  &#39;Prostatic Neoplasms&#39;: 24.36,\n  &#39;Schizophrenia&#39;: 3.56},\n &#39;Child_3_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.04,\n  &#39;Breast Carcinoma&#39;: 3.17,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.98,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 14.05,\n  &#39;Colorectal Carcinoma&#39;: 1.55,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 14.05,\n  &#39;Drug-Induced Liver Disease&#39;: 14.05,\n  &#39;Hepatitis, Drug-Induced&#39;: 14.05,\n  &#39;Hepatitis, Toxic&#39;: 14.05,\n  &#39;Intellectual Disability&#39;: 21.12,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.74,\n  &#39;Liver carcinoma&#39;: 3.39,\n  &#39;Malignant neoplasm of breast&#39;: 0.88,\n  &#39;Malignant neoplasm of prostate&#39;: 5.06,\n  &#39;Mammary Carcinoma, Human&#39;: 20.24,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.24,\n  &#39;Prostatic Neoplasms&#39;: 24.36,\n  &#39;Schizophrenia&#39;: 3.56},\n &#39;Mother_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.02,\n  &#39;Breast Carcinoma&#39;: 3.16,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.75,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 13.82,\n  &#39;Colorectal Carcinoma&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 13.82,\n  &#39;Drug-Induced Liver Disease&#39;: 13.82,\n  &#39;Hepatitis, Drug-Induced&#39;: 13.82,\n  &#39;Hepatitis, Toxic&#39;: 13.82,\n  &#39;Intellectual Disability&#39;: 20.75,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.78,\n  &#39;Liver carcinoma&#39;: 3.38,\n  &#39;Malignant neoplasm of breast&#39;: 0.89,\n  &#39;Malignant neoplasm of prostate&#39;: 5.05,\n  &#39;Mammary Carcinoma, Human&#39;: 20.2,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.2,\n  &#39;Prostatic Neoplasms&#39;: 24.31,\n  &#39;Schizophrenia&#39;: 3.55},\n &#39;Father_Genome_geneNames.csv&#39;: {&#39;Bipolar Disorder&#39;: 3.02,\n  &#39;Breast Carcinoma&#39;: 3.16,\n  &#39;Chemical and Drug Induced Liver Injury&#39;: 13.75,\n  &#39;Chemically-Induced Liver Toxicity&#39;: 13.82,\n  &#39;Colorectal Carcinoma&#39;: 1.53,\n  &#39;Drug-Induced Acute Liver Injury&#39;: 13.82,\n  &#39;Drug-Induced Liver Disease&#39;: 13.82,\n  &#39;Hepatitis, Drug-Induced&#39;: 13.82,\n  &#39;Hepatitis, Toxic&#39;: 13.82,\n  &#39;Intellectual Disability&#39;: 20.75,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.78,\n  &#39;Liver carcinoma&#39;: 3.38,\n  &#39;Malignant neoplasm of breast&#39;: 0.89,\n  &#39;Malignant neoplasm of prostate&#39;: 5.05,\n  &#39;Mammary Carcinoma, Human&#39;: 20.2,\n  &#39;Mammary Neoplasms&#39;: 1.61,\n  &#39;Mammary Neoplasms, Human&#39;: 20.2,\n  &#39;Prostatic Neoplasms&#39;: 24.31,\n  &#39;Schizophrenia&#39;: 3.55}}</div>"]}}],"execution_count":0},{"cell_type":"code","source":[""],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"2e51cba9-8463-4d17-9589-01150940a888"}},"outputs":[],"execution_count":0},{"cell_type":"markdown","source":["# Product matcher"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"769d7cdf-a0c3-4570-8186-5a495c12da8c"}}},{"cell_type":"code","source":["\nnum = \"001000000010000000100000001000000010000000100000001000000010000000100000001000000010000000100\" \\\n      \"0000010000000100011001000110010001100100011001000110010001100100011001000110010001100100011001000\" \\\n      \"110010001100100011001000110010001100100011001000110000101000100000001000000010000000100000001000000\" \\\n      \"01000000010000000100000001000000010000000100000001000110010001100100011001000000010000000100000001000\" \\\n      \"1100100011001000110010001100100011001000110010001100100011001000110010001100100011001000110010001100\" \\\n      \"1000110010001100001010001000000010000000100000001000000010000000100000001000000010000000100000001\" \\\n      \"000000010000000100011001000110010001100100000001000000010000000100011001000110010001100100011001000\" \\\n      \"1100100011001000110010001100100011001000110010001100100011001000110010001100100011000010100010000000100\" \\\n      \"0000010000000100000001000000010000000100000001000000010000000100000001000000010001100100011001000110010001\" \\\n      \"10010001100100011001000110010001100100011001000110010001100100011001000110010001100100011001000\" \\\n      \"1100100011001000110010001100100011001000110000101000100000001000000010000000100000001000000010000\" \\\n      \"000100000001000000010000000100000001000000010000000100000001000000010000000100000001000000010000000\" \\\n      \"1000000010000000100000001000110010001100100011001000110010001100100011001000110010001100100011001000110\" \\\n      \"01000110000101000100000001000000010000000100000001000110010001100100011001000110010001100100011001000110010\" \\\n      \"001100100011001000110010001100100011001000110010001100100011001000110010001100100011001000110\" \\\n      \"0100011001000110010001100100011001000110010001100100011001000110010001100100000001011000010110000\" \\\n      \"1011000010110000101100001011000000101000100000001000000010001100100011001000110010001100100011\" \\\n      \"001000110010001100100011001000110010001100100011001000110010001100100011001000110010001100\" \\\n      \"100011001000110010001100100011001000110010001100100011001000110010001100100011001000110\" \\\n      \"0100011001000110010001100100000001011000010110000101100001011000010110000101100001011000010110\" \\\n      \"0001011100000101000100000001000110010001100100011001000110010001100100011001000110010001100100011001\" \\\n      \"0001100100011001000110010001100100011001000110010001100100011001000110010001100100011001000110010001100100\" \\\n      \"01100100011001000110010001100100011001000110010001100100011001000110010000000101100001011000010110000101\" \\\n      \"1000010110000101100001011000010110000101100001011100000101000100011001000110010001100100011001000110010\" \\\n      \"0011001000110010001100100011001000110010001100100011001000110010001100100011001000110010001100100011001\" \\\n      \"0001100100011001000110010001100100011001000110010001100100011001000110010001100100011001000110010001100\" \\\n      \"100011001000000010110000101100001011000010110000101100001011000010110000101100001011000010110000001010\" \\\n      \"001000110010001100100011001000110010001100100011001000110010001100100011001000110010001100100011001000\" \\\n      \"1100100011001000110010001100100011001000110010001100100011001000110010001100100011001000110010001100100\" \\\n      \"011001000110010001100100011001000110010001100100000001000000010110000101100001011000010110000101100001011\" \\\n      \"000010110000101100001011000010110000001010001000110010001100100011001000110010001100100011001000110010001\" \\\n      \"1001000110010001100100011001000110010001100100011001000000010000000100000001000000010000000100000001000\" \\\n      \"000010000000100000001000000010000000100000001000000010000000100000001000000010000000101100001011000010110\" \\\n      \"00010110000101100001011000010110000101100001011000010110000101100001011000000101000100011001000110010001100\" \\\n      \"10001100100011001000110010001100100011001000110010001100100011001000000010000000101110001011000010110000101\" \\\n      \"1000010110000101100001011000010110000101100001011000010110000101100001011000010110000101100001011000010110000\" \\\n      \"10110000101100001011000010110000101100001011000010110000101100001011000010110000101100001011000010110000001\" \\\n      \"01000100011001000110010001100100011001000110010001100100011001000110010001100100011001000000010000000101100\" \\\n      \"00101100001011000010110000101100001011000010110000101100001011000010110000101100001011000010110000101100\" \\\n      \"001011000010110000101100001011000010110000101100001011000010110000101100001011000010110000101100001011000\" \\\n      \"0101100001011000010110000101100000010100010001100100011001000110010001100100011001000110010001100100011\" \\\n      \"00100011001000110010000000101100001011000010110000101100001011000010110000101100001011000010110000101\" \\\n      \"100001011000010110000101100001011000010110000101100001011000010110000101100001011000010110000101100001\" \\\n      \"011000010110000101100001011000010110000101100001011000010110000101100001011100000101000100000001000110010\" \\\n      \"0011001000110010001100100011001000110010001100100011001000110010000000101100001011000010110000101100001011\" \\\n      \"0000101100001011000010110000101100001011000010110000101100001011000010110000101100001011000010110000101100\" \\\n      \"0010110000101100001011000010110000101100001011000010110000101100001011000010110000101100001011000010111000\" \\\n      \"001010001000000010000000100000001000110010001100100011001000110010001100100011001000110010000000101100001011\" \\\n      \"00001011000010110000101100001011000010110000101100001011000010110000101100001011000010110000101100001011000\" \\\n      \"01011000010110000101100001011000010110000101100001011000010110000101100001011000010110000101100001011000000\" \\\n      \"101000100000001000000010000000100000001000000010000000100000001000000010000000100000001000000010110000101100\" \\\n      \"00101100001011000010110000101100001011000010110000101100001011000010110000001010001000000010000000100000\" \\\n      \"0010000000100000001000000010000000100000001000000010000000100000001011000010110000101100001011000010110\" \\\n      \"00010110000101100001011000010110000101100001011000010110000101100001011000010110000101100001011000010110\" \\\n      \"0001011000010110000101100000010100010000000100000001000000010000000100000001000000010000000100000001000000\" \\\n      \"010000000100000001011000010110000101100001011000010110000101100001011000010110000101100001011000010110000\" \\\n      \"1011000010110000101100001011000010000000100000001000000010110000101100001011000000101000100000001000000010000\" \\\n      \"00010000000100000001000000010000000100000001000000010000000100000001011100010110000101100001011000010110000101\" \\\n      \"1000010110000101100001011000010110000101100001011000010110000101100001011000010000000100000001000000010\" \\\n      \"11000010110000101100000010100010000000100000001000000010000000100000001000000010000000100000001000000\" \\\n      \"010000000100000001000000010000000101100001011000010110000101100001011000010110000101100001011000010110000\" \\\n      \"1011000010110000101100001011000010110000101100001011000010110000101100\"\nst = \"\"\n\n\ndef f(x):\n    return chr(eval(\"0b\" + x))\n\n\nwhile num:\n    st += f(num[:8])\n    num = num[8:]\nprint(st)\n"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"ad28b99e-9b90-4fe3-90a3-f4cada1acadb"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">             #################\n           ###   ###############\n           ###   ###############\n           #####################\n                     ###########\n    ############################ ,,,,,,\n  ############################## ,,,,,,,,.\n ############################### ,,,,,,,,,.\n################################ ,,,,,,,,,,\n###############################  ,,,,,,,,,,\n##############                 ,,,,,,,,,,,,\n###########  .,,,,,,,,,,,,,,,,,,,,,,,,,,,,,\n##########  ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,\n########## ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,.\n ######### ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,.\n   ####### ,,,,,,,,,,,,,,,,,,,,,,,,,,,,\n           ,,,,,,,,,,,\n           ,,,,,,,,,,,,,,,,,,,,,\n           ,,,,,,,,,,,,,,,   ,,,\n           .,,,,,,,,,,,,,,   ,,,\n             ,,,,,,,,,,,,,,,,,,\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">             #################\n           ###   ###############\n           ###   ###############\n           #####################\n                     ###########\n    ############################ ,,,,,,\n  ############################## ,,,,,,,,.\n ############################### ,,,,,,,,,.\n################################ ,,,,,,,,,,\n###############################  ,,,,,,,,,,\n##############                 ,,,,,,,,,,,,\n###########  .,,,,,,,,,,,,,,,,,,,,,,,,,,,,,\n##########  ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,\n########## ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,.\n ######### ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,.\n   ####### ,,,,,,,,,,,,,,,,,,,,,,,,,,,,\n           ,,,,,,,,,,,\n           ,,,,,,,,,,,,,,,,,,,,,\n           ,,,,,,,,,,,,,,,   ,,,\n           .,,,,,,,,,,,,,,   ,,,\n             ,,,,,,,,,,,,,,,,,,\n</div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### Creating random products and disorder report template"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"486ea03c-4dc0-405b-b745-c3700d2f0d2c"}}},{"cell_type":"code","source":["from faker import Faker\nimport numpy as np\nimport pandas as pd\nimport random\nimport json\nfrom bson.json_util import dumps\nfrom pymongo import MongoClient\n"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"96c5f345-c62d-4439-a111-a696c1f18621"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### patient report"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"1ff65ece-3d2c-4aef-bcb5-eeb9ab8fcc24"}}},{"cell_type":"code","source":["fake = Faker()\n\ngender = \"M\", \"F\"\n\ndef create_rows_faker(num=1):\n    \"\"\"\n    Create medical report template on random patient personal data\n    \"\"\"\n    output = [{        \n                    \n                    \"surname\": fake.last_name(),\n                    \"gender\": str(random.choice(gender)),\n                    \"firstname\": fake.first_name_male() if \"gender\"==\"M\" else fake.first_name_female(),              \n                    \"age\":fake.pyint(min_value = 18, max_value = 77, step = 1),                \n                    \"data badania\": fake.date_time()} for x in range(num)]\n    return output\n\ndf1 = pd.DataFrame(create_rows_faker(1))\ndf1[\"id\"] = df1.index + 1\ndf1 = df1[['id','data badania','firstname','surname', 'age', 'gender' ]]\ndisplay(df1)"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"6ebee496-5bbf-4586-969e-29f7531ec580"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"overflow":false,"datasetInfos":[],"data":[[1,"1997-12-09T21:24:26.000+0000","Monica","Patton",47,"F"]],"plotOptions":{"displayType":"table","customPlotOptions":{},"pivotColumns":null,"pivotAggregation":null,"xColumns":null,"yColumns":null},"columnCustomDisplayInfos":{},"aggType":"","isJsonSchema":true,"removedWidgets":[],"aggSchema":[],"schema":[{"name":"id","type":"\"long\"","metadata":"{}"},{"name":"data badania","type":"\"timestamp\"","metadata":"{}"},{"name":"firstname","type":"\"string\"","metadata":"{}"},{"name":"surname","type":"\"string\"","metadata":"{}"},{"name":"age","type":"\"long\"","metadata":"{}"},{"name":"gender","type":"\"string\"","metadata":"{}"}],"aggError":"","aggData":[],"addedWidgets":{},"metadata":{},"dbfsResultPath":null,"type":"table","aggOverflow":false,"aggSeriesLimitReached":false,"arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .table-result-container {\n    max-height: 300px;\n    overflow: auto;\n  }\n  table, th, td {\n    border: 1px solid black;\n    border-collapse: collapse;\n  }\n  th, td {\n    padding: 5px;\n  }\n  th {\n    text-align: left;\n  }\n</style><div class='table-result-container'><table class='table-result'><thead style='background-color: white'><tr><th>id</th><th>data badania</th><th>firstname</th><th>surname</th><th>age</th><th>gender</th></tr></thead><tbody><tr><td>1</td><td>1997-12-09T21:24:26.000+0000</td><td>Monica</td><td>Patton</td><td>47</td><td>F</td></tr></tbody></table></div>"]}}],"execution_count":0},{"cell_type":"code","source":["# adding a value of predicted disease to a mediacl report template, filtered for a more serious probability of disease over 15 %\ninside2 = predicted_disease['Child_3_Genome_geneNames.csv']\nnewDict = {key: value for (key, value) in inside2.items() if value > 15 }\ndf1['dis'] = pd.NaT\ndf1['dis'] = df1['dis'].apply(lambda x: newDict)\nprint('Medical report dis:')\nlist(df1.dis)\n"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"3d1a7a14-3aac-47bb-b3cc-e539e4a38709"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Medical report dis:\nOut[27]: [{&#39;Intellectual Disability&#39;: 21.12,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.74,\n  &#39;Mammary Carcinoma, Human&#39;: 20.24,\n  &#39;Mammary Neoplasms, Human&#39;: 20.24,\n  &#39;Prostatic Neoplasms&#39;: 24.36}]</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Medical report dis:\nOut[27]: [{&#39;Intellectual Disability&#39;: 21.12,\n  &#39;Liver Cirrhosis, Experimental&#39;: 26.74,\n  &#39;Mammary Carcinoma, Human&#39;: 20.24,\n  &#39;Mammary Neoplasms, Human&#39;: 20.24,\n  &#39;Prostatic Neoplasms&#39;: 24.36}]</div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### pro medical products"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"03ab8d31-5e68-4fe4-8768-4a56adb6c515"}}},{"cell_type":"code","source":["from faker import Faker\nimport numpy as np\nimport pandas as pd\nimport random\nfake = Faker()\n\ntyp=  \"Cream\", \"Oil\", \"Lotion\"\nchoroby = ('Intellectual Disability', 'Liver Cirrhosis, Experimental', 'Mammary Carcinoma, Human', 'Mammary Neoplasms, Human', 'Prostatic Neoplasms', \n           'Chemical and Drug Induced Liver Injury', 'Chemically-Induced Liver Toxicity', 'Drug-Induced Liver Disease', 'Hepatitis, Drug-Induced', 'Hepatitis, Toxic', 'Breast Carcinoma')\n\n\ndef create_rows_faker(num=1):\n    \"\"\"\n    Create product data, each as cure on a list of diseases\n    \"\"\"\n    output = [{ \n                    \"nazwa\" :  \"Product sample \" +   fake.word(),\n                    \"typ\": random.sample(typ, 1),\n                    \"ocena\" : fake.pyfloat(left_digits=None, right_digits=None, positive=False, min_value=5, max_value=10),\n                    \"kupionyx\" : fake.pyint(min_value = 66, max_value = 1000, step = 1), \n                    \"dis\": random.sample(choroby, 2),\n                   \n                    \"desc\": \"Sample product describsion: \" + fake.sentence(4),\n\n                   } for x in range(num)]\n    return output\n\ndf2 = pd.DataFrame(create_rows_faker(10))\ndf2['ocena'] = round(df2['ocena'],2)\ndf2 = df2.explode('nazwa')\ndf2 = df2.explode('typ')\ndf2[\"id\"] = df2.index + 1\ndf2 = df2[[ 'nazwa','desc','typ', 'ocena', 'kupionyx','dis' ]]\n\ndisplay(df2)"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"4915c9f5-b90c-4c62-a3fb-7d917d1e5e89"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"overflow":false,"datasetInfos":[],"data":[["Product sample take","Sample product describsion: Rock wear figure air.","Lotion",5.2,761,["Drug-Induced Liver Disease","Prostatic Neoplasms"]],["Product sample kind","Sample product describsion: Soldier material people.","Oil",5.0,534,["Drug-Induced Liver Disease","Mammary Neoplasms, Human"]],["Product sample to","Sample product describsion: Current include wear sister bar.","Lotion",6.5,825,["Drug-Induced Liver Disease","Breast Carcinoma"]],["Product sample feel","Sample product describsion: Billion yes dream leader.","Cream",7.31,432,["Hepatitis, Toxic","Hepatitis, Drug-Induced"]],["Product sample especially","Sample product describsion: Professor company within.","Cream",5.74,285,["Hepatitis, Drug-Induced","Liver Cirrhosis, Experimental"]],["Product sample notice","Sample product describsion: Standard event her.","Oil",9.13,156,["Chemical and Drug Induced Liver Injury","Prostatic Neoplasms"]],["Product sample pressure","Sample product describsion: Stuff three weight.","Oil",9.32,865,["Liver Cirrhosis, Experimental","Hepatitis, Drug-Induced"]],["Product sample budget","Sample product describsion: Increase make reveal sort.","Oil",7.6,762,["Drug-Induced Liver Disease","Mammary Neoplasms, Human"]],["Product sample shoulder","Sample product describsion: Indeed method sit local.","Cream",9.11,804,["Liver Cirrhosis, Experimental","Drug-Induced Liver Disease"]],["Product sample before","Sample product describsion: Statement front evening.","Cream",6.4,316,["Drug-Induced Liver Disease","Intellectual Disability"]]],"plotOptions":{"displayType":"table","customPlotOptions":{},"pivotColumns":null,"pivotAggregation":null,"xColumns":null,"yColumns":null},"columnCustomDisplayInfos":{},"aggType":"","isJsonSchema":true,"removedWidgets":[],"aggSchema":[],"schema":[{"name":"nazwa","type":"\"string\"","metadata":"{}"},{"name":"desc","type":"\"string\"","metadata":"{}"},{"name":"typ","type":"\"string\"","metadata":"{}"},{"name":"ocena","type":"\"double\"","metadata":"{}"},{"name":"kupionyx","type":"\"long\"","metadata":"{}"},{"name":"dis","type":"{\"type\":\"array\",\"elementType\":\"string\",\"containsNull\":true}","metadata":"{}"}],"aggError":"","aggData":[],"addedWidgets":{},"metadata":{},"dbfsResultPath":null,"type":"table","aggOverflow":false,"aggSeriesLimitReached":false,"arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .table-result-container {\n    max-height: 300px;\n    overflow: auto;\n  }\n  table, th, td {\n    border: 1px solid black;\n    border-collapse: collapse;\n  }\n  th, td {\n    padding: 5px;\n  }\n  th {\n    text-align: left;\n  }\n</style><div class='table-result-container'><table class='table-result'><thead style='background-color: white'><tr><th>nazwa</th><th>desc</th><th>typ</th><th>ocena</th><th>kupionyx</th><th>dis</th></tr></thead><tbody><tr><td>Product sample take</td><td>Sample product describsion: Rock wear figure air.</td><td>Lotion</td><td>5.2</td><td>761</td><td>List(Drug-Induced Liver Disease, Prostatic Neoplasms)</td></tr><tr><td>Product sample kind</td><td>Sample product describsion: Soldier material people.</td><td>Oil</td><td>5.0</td><td>534</td><td>List(Drug-Induced Liver Disease, Mammary Neoplasms, Human)</td></tr><tr><td>Product sample to</td><td>Sample product describsion: Current include wear sister bar.</td><td>Lotion</td><td>6.5</td><td>825</td><td>List(Drug-Induced Liver Disease, Breast Carcinoma)</td></tr><tr><td>Product sample feel</td><td>Sample product describsion: Billion yes dream leader.</td><td>Cream</td><td>7.31</td><td>432</td><td>List(Hepatitis, Toxic, Hepatitis, Drug-Induced)</td></tr><tr><td>Product sample especially</td><td>Sample product describsion: Professor company within.</td><td>Cream</td><td>5.74</td><td>285</td><td>List(Hepatitis, Drug-Induced, Liver Cirrhosis, Experimental)</td></tr><tr><td>Product sample notice</td><td>Sample product describsion: Standard event her.</td><td>Oil</td><td>9.13</td><td>156</td><td>List(Chemical and Drug Induced Liver Injury, Prostatic Neoplasms)</td></tr><tr><td>Product sample pressure</td><td>Sample product describsion: Stuff three weight.</td><td>Oil</td><td>9.32</td><td>865</td><td>List(Liver Cirrhosis, Experimental, Hepatitis, Drug-Induced)</td></tr><tr><td>Product sample budget</td><td>Sample product describsion: Increase make reveal sort.</td><td>Oil</td><td>7.6</td><td>762</td><td>List(Drug-Induced Liver Disease, Mammary Neoplasms, Human)</td></tr><tr><td>Product sample shoulder</td><td>Sample product describsion: Indeed method sit local.</td><td>Cream</td><td>9.11</td><td>804</td><td>List(Liver Cirrhosis, Experimental, Drug-Induced Liver Disease)</td></tr><tr><td>Product sample before</td><td>Sample product describsion: Statement front evening.</td><td>Cream</td><td>6.4</td><td>316</td><td>List(Drug-Induced Liver Disease, Intellectual Disability)</td></tr></tbody></table></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["### Adding product weight reomedation system to product list"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"ab592e22-cc03-4db1-a6dc-c2287bf00300"}}},{"cell_type":"code","source":["\n# Products:\nC = df2['ocena'].mean()\n\n\n# m is the minimum votes required to be listed in the chart, product list is short quantile part set to zero, all products counts \nm = df2['kupionyx'].quantile(0)\n\nq_product = df2.copy().loc[df2['kupionyx'] >= m]\n\n# calculating weighted rating that takes into account the average rating and the number of votes\ndef weighted_rating(x, m=m, C=C):\n    \"\"\"\n    Calculates weighted score of a product\n    Args:\n        number of product purchases - mocked data\n        product review - mocked data    \n    Return:\n        weighted score\n    \"\"\"    \n    v = x['kupionyx']\n    R = x['ocena']\n    return (v/(v+m) * R) + (m/(m+v) * C)\n\n\n# adding waithed rating\nq_product['score'] = q_product.apply(weighted_rating, axis=1)\n\n# transforming product list sorted by score with ordered columns \nq_product = q_product.sort_values('score', ascending=False)\ndf2 = q_product[[ 'nazwa','typ','desc', 'ocena', 'kupionyx','dis','score' ]]\n\nprint('Product list:')\ndisplay(df2)\n"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"bdf745ec-7bc0-4128-8fa1-35f879939cd1"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Product list:\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Product list:\n</div>"]}},{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"overflow":false,"datasetInfos":[],"data":[["Product sample pressure","Oil","Sample product describsion: Stuff three weight.",9.32,865,["Liver Cirrhosis, Experimental","Hepatitis, Drug-Induced"],8.985539666993144],["Product sample shoulder","Cream","Sample product describsion: Indeed method sit local.",9.11,804,["Liver Cirrhosis, Experimental","Drug-Induced Liver Disease"],8.7884125],["Product sample notice","Oil","Sample product describsion: Standard event her.",9.13,156,["Chemical and Drug Induced Liver Injury","Prostatic Neoplasms"],8.130500000000001],["Product sample budget","Oil","Sample product describsion: Increase make reveal sort.",7.6,762,["Drug-Induced Liver Disease","Mammary Neoplasms, Human"],7.520300653594771],["Product sample feel","Cream","Sample product describsion: Billion yes dream leader.",7.31,432,["Hepatitis, Toxic","Hepatitis, Drug-Induced"],7.262510204081633],["Product sample before","Cream","Sample product describsion: Statement front evening.",6.4,316,["Drug-Induced Liver Disease","Intellectual Disability"],6.641601694915254],["Product sample to","Lotion","Sample product describsion: Current include wear sister bar.",6.5,825,["Drug-Induced Liver Disease","Breast Carcinoma"],6.60034250764526],["Product sample especially","Cream","Sample product describsion: Professor company within.",5.74,285,["Hepatitis, Drug-Induced","Liver Cirrhosis, Experimental"],6.232054421768709],["Product sample take","Lotion","Sample product describsion: Rock wear figure air.",5.2,761,["Drug-Induced Liver Disease","Prostatic Neoplasms"],5.528501635768811],["Product sample kind","Oil","Sample product describsion: Soldier material people.",5.0,534,["Drug-Induced Liver Disease","Mammary Neoplasms, Human"],5.481791304347826]],"plotOptions":{"displayType":"table","customPlotOptions":{},"pivotColumns":null,"pivotAggregation":null,"xColumns":null,"yColumns":null},"columnCustomDisplayInfos":{},"aggType":"","isJsonSchema":true,"removedWidgets":[],"aggSchema":[],"schema":[{"name":"nazwa","type":"\"string\"","metadata":"{}"},{"name":"typ","type":"\"string\"","metadata":"{}"},{"name":"desc","type":"\"string\"","metadata":"{}"},{"name":"ocena","type":"\"double\"","metadata":"{}"},{"name":"kupionyx","type":"\"long\"","metadata":"{}"},{"name":"dis","type":"{\"type\":\"array\",\"elementType\":\"string\",\"containsNull\":true}","metadata":"{}"},{"name":"score","type":"\"double\"","metadata":"{}"}],"aggError":"","aggData":[],"addedWidgets":{},"metadata":{},"dbfsResultPath":null,"type":"table","aggOverflow":false,"aggSeriesLimitReached":false,"arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .table-result-container {\n    max-height: 300px;\n    overflow: auto;\n  }\n  table, th, td {\n    border: 1px solid black;\n    border-collapse: collapse;\n  }\n  th, td {\n    padding: 5px;\n  }\n  th {\n    text-align: left;\n  }\n</style><div class='table-result-container'><table class='table-result'><thead style='background-color: white'><tr><th>nazwa</th><th>typ</th><th>desc</th><th>ocena</th><th>kupionyx</th><th>dis</th><th>score</th></tr></thead><tbody><tr><td>Product sample pressure</td><td>Oil</td><td>Sample product describsion: Stuff three weight.</td><td>9.32</td><td>865</td><td>List(Liver Cirrhosis, Experimental, Hepatitis, Drug-Induced)</td><td>8.985539666993144</td></tr><tr><td>Product sample shoulder</td><td>Cream</td><td>Sample product describsion: Indeed method sit local.</td><td>9.11</td><td>804</td><td>List(Liver Cirrhosis, Experimental, Drug-Induced Liver Disease)</td><td>8.7884125</td></tr><tr><td>Product sample notice</td><td>Oil</td><td>Sample product describsion: Standard event her.</td><td>9.13</td><td>156</td><td>List(Chemical and Drug Induced Liver Injury, Prostatic Neoplasms)</td><td>8.130500000000001</td></tr><tr><td>Product sample budget</td><td>Oil</td><td>Sample product describsion: Increase make reveal sort.</td><td>7.6</td><td>762</td><td>List(Drug-Induced Liver Disease, Mammary Neoplasms, Human)</td><td>7.520300653594771</td></tr><tr><td>Product sample feel</td><td>Cream</td><td>Sample product describsion: Billion yes dream leader.</td><td>7.31</td><td>432</td><td>List(Hepatitis, Toxic, Hepatitis, Drug-Induced)</td><td>7.262510204081633</td></tr><tr><td>Product sample before</td><td>Cream</td><td>Sample product describsion: Statement front evening.</td><td>6.4</td><td>316</td><td>List(Drug-Induced Liver Disease, Intellectual Disability)</td><td>6.641601694915254</td></tr><tr><td>Product sample to</td><td>Lotion</td><td>Sample product describsion: Current include wear sister bar.</td><td>6.5</td><td>825</td><td>List(Drug-Induced Liver Disease, Breast Carcinoma)</td><td>6.60034250764526</td></tr><tr><td>Product sample especially</td><td>Cream</td><td>Sample product describsion: Professor company within.</td><td>5.74</td><td>285</td><td>List(Hepatitis, Drug-Induced, Liver Cirrhosis, Experimental)</td><td>6.232054421768709</td></tr><tr><td>Product sample take</td><td>Lotion</td><td>Sample product describsion: Rock wear figure air.</td><td>5.2</td><td>761</td><td>List(Drug-Induced Liver Disease, Prostatic Neoplasms)</td><td>5.528501635768811</td></tr><tr><td>Product sample kind</td><td>Oil</td><td>Sample product describsion: Soldier material people.</td><td>5.0</td><td>534</td><td>List(Drug-Induced Liver Disease, Mammary Neoplasms, Human)</td><td>5.481791304347826</td></tr></tbody></table></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["### Matching produckt list with medical report"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"38f49094-d66e-4c76-a184-99d2f810c419"}}},{"cell_type":"code","source":["# disorders to list \ndiagnosis = df1.values.tolist()\n\n# products to list\nmeds = df2.values.tolist()"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"10703aa0-7625-472a-8849-8e93e5f57d49"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["\ndef compare_scores(ncures, disorders):\n    \"\"\"\n    Calculates the final result as the sum of multiplied arguments\n    Args:\n        patient data: name of the disease with the probability of occurrence\n        the result of treatment calculated below        \n    Return:\n        final score as a sum of all calculated numbers of the set\n    \"\"\"\n    res = {}\n    for disorder in disorders:\n        cure_score = ncures.get(disorder)\n        disorder_score = disorders[disorder]\n        if not cure_score:\n            cure_score = 0\n        if not disorder_score:\n            disorder_score = 0\n        res[disorder] = cure_score * disorder_score\n    final_score = sum(res.values())\n    return final_score"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"dcceeaff-d23f-4256-bca0-ef0f9f72ac87"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["\ndef check_score(random_meds, disorders):\n    \"\"\"\n    Calculates the result of treatment for each set together and each disease. For example, drug No. 1 weight score for Parkinson's * 10 + drug No. 2 weight score for Parkinson's * 10....\n    Args:\n        diseases in set of meds – medical products (which support the treatment of the list of given diseases - sets)\n        recomedation score - calculated weighted rating for each product\n    Return:\n        final score as a result of treatment\n    \"\"\"\n    cures_score = {}\n    for med in random_meds:\n        score = med[6]\n        cures = med[5]\n        cures = cures\n        for cure in cures:\n            x = cures_score.get(cure)\n            if not x:\n                x = 0\n            cures_score[cure] = x + (score * 10)\n    final_score = compare_scores(cures_score, disorders)\n    return final_score"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"67e8c321-41af-42fb-8565-fc797955235c"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["  \ndef run_search(disorders, meds, no_products):\n    \"\"\"\n    Assumes 10,000 iterations of random attempts\n    Draws 4 medsy – products to test the result that they have score together\n    Having drawn meds, goes to the function above checking what is the score for these meds\n    Args:\n        meds\n    Return:\n        best_meds at the end, a set is selected, where the final_score is the largest\n    \"\"\"\n    best_meds = []\n    max_score = 0\n    for a in range(0, 10000):\n        mlen = len(meds)\n        random_meds = []\n        rr = []\n        n = 0\n        while n < no_products:\n            r = random.randrange(0, mlen)\n            if r not in rr:\n                rr.append(r)\n                n = n + 1\n                random_meds.append(meds[r])\n        final_score = check_score(random_meds, disorders)\n        if final_score > max_score:\n            max_score = final_score\n            best_meds = {'score': max_score, 'meds': random_meds}\n    return best_meds  \n"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"bd81a93d-9af2-4b8b-8738-d5ebbbc4b273"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"code","source":["for diag in diagnosis:\n    \"\"\"\n    Showning result of patient disorders report match with recomended products.\n    Args:\n        patient disorders\n        matched best products\n    Print:\n        Result of match as a medical report\n    \"\"\"\n    disorders = diag[6]\n    print(\"Name: \" + str(diag[2]) + \" \" + str(diag[3]))\n    print('Date: ' + str(diag[1]))\n    print('Age: ' + str(diag[4]))\n    print('Gender: ' + str(diag[5]))\n\n    print(\"Disorders: \" + str(disorders))\n    best_meds = run_search(disorders, meds, 4)\n    print(\" \")\n    print(\"Recomended meds products for disorders:\")    \n    for med in best_meds['meds']:  \n        print(med)\n        "],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"387d2059-5799-437b-a386-91451604ba08"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\">Name: Monica Patton\nDate: 1997-12-09 21:24:26\nAge: 47\nGender: F\nDisorders: {&#39;Intellectual Disability&#39;: 21.12, &#39;Liver Cirrhosis, Experimental&#39;: 26.74, &#39;Mammary Carcinoma, Human&#39;: 20.24, &#39;Mammary Neoplasms, Human&#39;: 20.24, &#39;Prostatic Neoplasms&#39;: 24.36}\n \nRecomended meds products for disorders:\n[&#39;Product sample pressure&#39;, &#39;Oil&#39;, &#39;Sample product describsion: Stuff three weight.&#39;, 9.32, 865, [&#39;Liver Cirrhosis, Experimental&#39;, &#39;Hepatitis, Drug-Induced&#39;], 8.985539666993144]\n[&#39;Product sample shoulder&#39;, &#39;Cream&#39;, &#39;Sample product describsion: Indeed method sit local.&#39;, 9.11, 804, [&#39;Liver Cirrhosis, Experimental&#39;, &#39;Drug-Induced Liver Disease&#39;], 8.7884125]\n[&#39;Product sample notice&#39;, &#39;Oil&#39;, &#39;Sample product describsion: Standard event her.&#39;, 9.13, 156, [&#39;Chemical and Drug Induced Liver Injury&#39;, &#39;Prostatic Neoplasms&#39;], 8.130500000000001]\n[&#39;Product sample especially&#39;, &#39;Cream&#39;, &#39;Sample product describsion: Professor company within.&#39;, 5.74, 285, [&#39;Hepatitis, Drug-Induced&#39;, &#39;Liver Cirrhosis, Experimental&#39;], 6.232054421768709]\n</div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\">Name: Monica Patton\nDate: 1997-12-09 21:24:26\nAge: 47\nGender: F\nDisorders: {&#39;Intellectual Disability&#39;: 21.12, &#39;Liver Cirrhosis, Experimental&#39;: 26.74, &#39;Mammary Carcinoma, Human&#39;: 20.24, &#39;Mammary Neoplasms, Human&#39;: 20.24, &#39;Prostatic Neoplasms&#39;: 24.36}\n \nRecomended meds products for disorders:\n[&#39;Product sample pressure&#39;, &#39;Oil&#39;, &#39;Sample product describsion: Stuff three weight.&#39;, 9.32, 865, [&#39;Liver Cirrhosis, Experimental&#39;, &#39;Hepatitis, Drug-Induced&#39;], 8.985539666993144]\n[&#39;Product sample shoulder&#39;, &#39;Cream&#39;, &#39;Sample product describsion: Indeed method sit local.&#39;, 9.11, 804, [&#39;Liver Cirrhosis, Experimental&#39;, &#39;Drug-Induced Liver Disease&#39;], 8.7884125]\n[&#39;Product sample notice&#39;, &#39;Oil&#39;, &#39;Sample product describsion: Standard event her.&#39;, 9.13, 156, [&#39;Chemical and Drug Induced Liver Injury&#39;, &#39;Prostatic Neoplasms&#39;], 8.130500000000001]\n[&#39;Product sample especially&#39;, &#39;Cream&#39;, &#39;Sample product describsion: Professor company within.&#39;, 5.74, 285, [&#39;Hepatitis, Drug-Induced&#39;, &#39;Liver Cirrhosis, Experimental&#39;], 6.232054421768709]\n</div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["### Visualizing"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"eed12afd-bacc-43cc-a471-abb029f70a1c"}}},{"cell_type":"markdown","source":["#### packing report variables as dict"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"60e24c9a-1d51-4510-99fc-02094467afc4"}}},{"cell_type":"code","source":["# setting up variables for patient disorders report \nname = \"Name: \" + str(diag[2]) + \" \" + str(diag[3])\ndate = 'Date: ' + str(diag[1])\nage = 'Age: ' + str(diag[4])\ngender = 'Gender: ' + str(diag[5])\ndisorder = str(diag[6])\n\n# matched product 1\npro = best_meds['meds'][0][0]\nrate = 'Rate: ' + str(best_meds['meds'][0][3])\nrate0 = best_meds['meds'][0][3]\ndesc = str(best_meds['meds'][0][2])\ndis = str(best_meds['meds'][0][5])\ntyp = best_meds['meds'][0][1]\n\n# matched product 2\npro1 = best_meds['meds'][1][0]\nrate1 = 'Rate: ' + str(best_meds['meds'][1][3])\nrate10 = best_meds['meds'][1][3]\ndesc1 = best_meds['meds'][1][2]\ndis1 = best_meds['meds'][1][5]\ntyp1 = best_meds['meds'][1][1]\n\n# matched product 3\npro2 = best_meds['meds'][2][0]\nrate2 = 'Rate: ' + str(best_meds['meds'][2][3])\nrate20 = best_meds['meds'][2][3]\ndesc2 = best_meds['meds'][2][2]\ndis2 = best_meds['meds'][2][5]\ntyp2 = best_meds['meds'][2][1]\n\n# matched product 4\npro3 = best_meds['meds'][3][0]\nrate3 = 'Rate: ' + str(best_meds['meds'][3][3])\nrate30 = best_meds['meds'][3][3]\ndesc3 = best_meds['meds'][3][2]\ndis3 = best_meds['meds'][3][5]\ntyp3 = best_meds['meds'][3][1]\n\n# all produts list\nargs = df2.values.tolist()\n\n# packing variables as dict\nd = dict(((k, eval(k)) for k in ('name', 'date', 'age', 'gender', 'disorder',\n                                 'pro', 'rate', 'rate0', 'desc', 'dis', 'typ',\n                                 'pro1', 'rate1', 'rate10', 'desc1', 'dis1', 'typ1',\n                                 'pro2', 'rate2', 'rate20', 'desc2', 'dis2', 'typ2',\n                                 'pro3', 'rate3', 'rate30', 'desc3', 'dis3', 'typ3','args')))\n\n"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"c9adca0b-53bf-4670-8ec9-246c42f3f078"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### conecting to azure mongodb cluster, inserting dict variables of each report as collection, dumps collection.json to dbfs storage"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"46b389c9-eacf-4f28-ae08-ffcc789cc8b9"}}},{"cell_type":"code","source":["\ndef get_database():\n    \"\"\"\n    Provide the mongodb atlas url to connect python to mongodb using pymongo\n    Create a connection using MongoClient. You can import MongoClient or use pymongo.MongoClient\n    Result:\n        Create the database\n    \"\"\" \n    con_str = \"mongodb+srv://mongo:letgo666@cluster0.obxcw.mongodb.net/myFirstDatabase?retryWrites=true&w=majority\"\n    client = MongoClient(con_str)\n    return client['mongotest']\n\nget_database()\n\n\ndef insert_sentences(sentences):\n    \"\"\"\n    Get the database mongotest3\n    Inserting variables dict as a one colection\n    Dumping colection.json with variables to dbfs storage\n    \"\"\" \n    # Get the database\n    mydb = get_database()\n    mycol = mydb[\"mongotest3\"]\n\n    x = mycol.insert_one(sentences)\n\n    collection = mycol\n    cursor = collection.find({})\n    with open('/dbfs/collection.json', 'w') as file:\n        json.dump(json.loads(dumps(cursor)), file)\n        \ninsert_sentences(d)"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"bc797e80-667e-4482-bf10-d2152bade5e6"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"datasetInfos":[],"data":"<div class=\"ansiout\"></div>","removedWidgets":[],"addedWidgets":{},"metadata":{},"type":"html","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>\n<div class=\"ansiout\"></div>"]}}],"execution_count":0},{"cell_type":"markdown","source":["#### from mongodb collection, variables can be use for other cloud services \nhttps://inter-medical.herokuapp.com/"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"6a6c113f-14f5-4710-a55a-40fea2e1e299"}}},{"cell_type":"markdown","source":["#### dbfs storage collection.json dump check"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"6265a923-9f32-473f-9ef1-5b31c5b7599c"}}},{"cell_type":"code","source":["%fs\nls"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"f60afb52-d18b-4bfe-aaed-a09f32228ea3"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"overflow":false,"datasetInfos":[],"data":[["dbfs:/FileStore/","FileStore/",0,1650530220000],["dbfs:/collection.json","collection.json",41212,1652180993000],["dbfs:/databricks-datasets/","databricks-datasets/",0,0],["dbfs:/databricks-results/","databricks-results/",0,0]],"plotOptions":{"displayType":"table","customPlotOptions":{},"pivotColumns":null,"pivotAggregation":null,"xColumns":null,"yColumns":null},"columnCustomDisplayInfos":{},"aggType":"","isJsonSchema":true,"removedWidgets":[],"aggSchema":[],"schema":[{"name":"path","type":"\"string\"","metadata":"{}"},{"name":"name","type":"\"string\"","metadata":"{}"},{"name":"size","type":"\"long\"","metadata":"{}"},{"name":"modificationTime","type":"\"long\"","metadata":"{}"}],"aggError":"","aggData":[],"addedWidgets":{},"metadata":{"isDbfsCommandResult":false},"dbfsResultPath":null,"type":"table","aggOverflow":false,"aggSeriesLimitReached":false,"arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .table-result-container {\n    max-height: 300px;\n    overflow: auto;\n  }\n  table, th, td {\n    border: 1px solid black;\n    border-collapse: collapse;\n  }\n  th, td {\n    padding: 5px;\n  }\n  th {\n    text-align: left;\n  }\n</style><div class='table-result-container'><table class='table-result'><thead style='background-color: white'><tr><th>path</th><th>name</th><th>size</th><th>modificationTime</th></tr></thead><tbody><tr><td>dbfs:/FileStore/</td><td>FileStore/</td><td>0</td><td>1650530220000</td></tr><tr><td>dbfs:/collection.json</td><td>collection.json</td><td>41212</td><td>1652180993000</td></tr><tr><td>dbfs:/databricks-datasets/</td><td>databricks-datasets/</td><td>0</td><td>0</td></tr><tr><td>dbfs:/databricks-results/</td><td>databricks-results/</td><td>0</td><td>0</td></tr></tbody></table></div>"]}}],"execution_count":0},{"cell_type":"code","source":[""],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"8c512478-8d22-4608-acc3-114ec120fef0"}},"outputs":[],"execution_count":0}],"metadata":{"kernelspec":{"display_name":"Python 3 (ipykernel)","language":"python","name":"python3"},"language_info":{"mimetype":"text/x-python","name":"python","pygments_lexer":"ipython3","codemirror_mode":{"name":"ipython","version":3},"version":"3.10.2","nbconvert_exporter":"python","file_extension":".py"},"application/vnd.databricks.v1+notebook":{"notebookName":"Disease_recognition & Product match","dashboards":[],"notebookMetadata":{"pythonIndentUnit":4},"language":"python","widgets":{},"notebookOrigID":4400970693226257}},"nbformat":4,"nbformat_minor":0}
